game-core/src/main/java/games/strategy/triplea/ui/logic/MapScrollUtil.java

```diff
@@ -0,0 +1,57 @@
+package games.strategy.triplea.ui.logic;
+
+import java.awt.geom.AffineTransform;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+
+/**
+ * Helper class to deal with infinitely scrolling maps.
+ */
+public class MapScrollUtil {
+  /**
+   * Creates an unmodifiable List of possible translations on the given map.
+   * If a map has "fixed borders" i.e. you can't infinitely scroll along at least one axis
+   * the returned list will just contain a single identity {@linkplain AffineTransform}.
+   *
+   * <p>
+   * If the map however is infinitely scrolling along an axis, the amount of {@linkplain AffineTransform}s
+   * will multiply by 3.
+   * Each {@linkplain AffineTransform} is a translation by a multiple (between -1 and 1) of mapHeight/mapWidth.
+   * </p>
+   *
+   * @param isInfiniteX True if the map wraps left and right
+   * @param isInfiniteY True if the map wraps top and bottom
+   * @param mapWidth Unscaled width of the map
+   * @param mapHeight Unscaled height of the map
+   *
+   * @return An unmodifiable List containing 9-1 {@linkplain AffineTransform}s
+   */
+  public static List<AffineTransform> getPossibleTranslations(
```

Comment:
#3725 removed unit tests for this method (in `MapScrollUtilTest`).  Do you not want to add them back?